### PR TITLE
Use arrays for put/delete of ssh keys and option to add vs set.

### PR DESCRIPTION
**Pull request by fabricematrat**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/idmclient/60/
Test PASSed.

File: params/params.go [hunk 1]
```diff
@@ -139,33 +139,38 @@ type VerifyTokenRequest struct {
 	Macaroons         macaroon.Slice `httprequest:",body"`
 }
 
-// UserSSHKeysRequest is a request for the list of ssh keys associated
+// GetSSHKeysRequest is a request for the list of ssh keys associated
 // with the specified user.
-type UserSSHKeysRequest struct {
+type GetSSHKeysRequest struct {
```
Comment (mhilton) [inline params/params.go]: perhaps just SSHKeysRequest?

Comment (rogpeppe) [inline params/params.go]: SSHKeysRequest? (to be consistent with SSHKeysResponse).

Most of the other GET requests don't have a "Get" prefix.

**Comment by mhilton** [comment]
:+1: with a thought

File: params/params.go [hunk 2]
```diff
@@ -139,33 +139,38 @@ type VerifyTokenRequest struct {
...
+// associated with the user.
+type PutSSHKeysRequest struct {
 	httprequest.Route `httprequest:"PUT /v1/u/:username/ssh-keys"`
-	Username          Username `httprequest:"username,path"`
-	SSHKey            string   `httprequest:",body"`
+	Username          Username       `httprequest:"username,path"`
+	Body              PutSSHKeysBody `httprequest:",body"`
+}
+
+type PutSSHKeysBody struct {
```
Comment (rogpeppe) [inline params/params.go]: Doc comments please.

File: params/params.go [hunk 3]
```diff
@@ -139,33 +139,38 @@ type VerifyTokenRequest struct {
...
+	SSHKeys           []string
+	Add               bool
```
Comment (rogpeppe) [inline params/params.go]: `json:"add,omitempty"`
?

File: params/params.go [hunk 4]
```diff
@@ -139,33 +139,38 @@ type VerifyTokenRequest struct {
...
+type PutSSHKeysRequest struct {
 	httprequest.Route `httprequest:"PUT /v1/u/:username/ssh-keys"`
-	Username          Username `httprequest:"username,path"`
-	SSHKey            string   `httprequest:",body"`
+	Username          Username       `httprequest:"username,path"`
+	Body              PutSSHKeysBody `httprequest:",body"`
+}
+
+type PutSSHKeysBody struct {
+	SSHKeys           []string
```
Comment (rogpeppe) [inline params/params.go]: `json:"ssh-keys"`

?

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/idmclient/61/
Test PASSed.

File: params/params.go [hunk 5]
```diff
@@ -139,33 +139,38 @@ type VerifyTokenRequest struct {
...
-// RemoveSSHKeyRequest is a request to remove an ssh key to the list of ssh keys
-// associated with the current user.
-type RemoveSSHKeyRequest struct {
+// DeleteSSHKeysRequest is a request to remove ssh keys from the list of ssh keys
+// associated with the user.
+type DeleteSSHKeysRequest struct {
 	httprequest.Route `httprequest:"DELETE /v1/u/:username/ssh-keys"`
 	Username          Username `httprequest:"username,path"`
-	SSHKey            string   `httprequest:",body"`
+	SSHKeys           []string `httprequest:",body"`
```
Comment (rogpeppe) [inline params/params.go]: This should probably be an object not a literal slice.

**Comment by rogpeppe** [comment]
LGTM modulo the above suggestions, thanks.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/idmclient/62/
Test PASSed.

**Comment by fabricematrat** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/idmclient-merge

**Pull request merged**